types/json/binary.go

```diff
@@ -551,9 +558,15 @@ func appendBinaryNumber(buf []byte, x json.Number) (TypeCode, []byte, error) {
 		typeCode = TypeCodeInt64
 		i64, err := x.Int64()
 		if err != nil {
-			return typeCode, nil, errors.Trace(err)
+			typeCode = TypeCodeFloat64
+			f64, err := x.Float64()
```

Comment:
Is this a bug fix?

Reply:
Yes, to pass the tests in `expression` package.